pkg/handlers/archive_test.go

```diff
@@ -183,3 +188,84 @@ func TestExtractRPMContent(t *testing.T) {
 	expectedLength := 1822720
 	assert.Equal(t, expectedLength, len(string(content)))
 }
+
+func TestFoundKeyInDockerTar(t *testing.T) {
+	// URI of the Docker image to clone
+	imageURI := "ghcr.io/joeleonjr/getting-started-app-with-canary-token:main"
+
+	var imageName name.Reference
+	imageName, err := name.NewTag(imageURI)
+	if err != nil {
+		t.Errorf("%s: error creating image name: %s", imageURI, err)
+	}
+
+	// Pull the image
+	img, err := remote.Image(imageName)
+	if err != nil {
+		t.Errorf("%s: error pulling image: %s", imageURI, err)
+	}
+
+	tempFile, err := os.CreateTemp("", "archive_test_docker_img.tar")
+	if err != nil {
+		t.Errorf("%s: error creating temporary file: %s", imageURI, err)
+		return
+	}
+
+	defer os.Remove(tempFile.Name()) // Clean up the temporary file
+
+	// Save the image as a tar file
+	err = tarball.WriteToFile(tempFile.Name(), imageName, img)
+	if err != nil {
+		t.Errorf("%s: error saving image as tar file: %s", imageURI, err)
+		return
+	}
+
+	// ctx := logContext.AddLogger(context.Background())
```

Comment:
You almost had it! The reason why this test wasn't completing is because you wan't to put `HandleFile` in a go routine while at the same time reading from the chunksChan. That would also simulate how this would work normally. We have sources running concurrently with the detection engine. The sources will put chunks into the chunks chan and the workers will pull chunks out and handle detection, notification, etc. So to make this work you can do:
```
ctx := logContext.AddLogger(context.Background())

	inputFile, err := os.Open(tempFile.Name())
	if err != nil {
		t.Errorf("%s: error opening tar file: %s", imageURI, err)
		return
	}
	defer inputFile.Close()

	reReader, err := diskbufferreader.New(inputFile)
	if err != nil {
		t.Errorf("%s: error creating re-readable reader: %s", imageURI, err)
		return
	}
	defer reReader.Close()

	chunkSkel := &sources.Chunk{
		SourceType: 1,
		SourceName: "filesystem",
		SourceID:   1,
		SourceMetadata: &source_metadatapb.MetaData{
			Data: &source_metadatapb.MetaData_Filesystem{
				Filesystem: &source_metadatapb.Filesystem{
					File: sanitizer.UTF8(tempFile.Name()),
				},
			},
		},
		Verify: true,
	}

	chunksChan := make(chan *sources.Chunk, 1)

	go func() {
		defer close(chunksChan)
		HandleFile(ctx, reReader, chunkSkel, chunksChan)
	}()

	secret := "AKIA2OGYBAH6Q2PQJUGN"

	// Read from the channel and validate the secrets.
	foundSecret := ""
	for chunkCh := range chunksChan {
		foundSecret += string(chunkCh.Data)
	}

	assert.Contains(t, foundSecret, secret)
```

Reply:
So close! Thank you for adding that in. I also separated this into an `archive_integration_test.go` file instead.